agent/stats/engine.go

```diff
@@ -229,58 +258,110 @@ func (engine *DockerStatsEngine) removeAll() {
 }
 
 // addContainerUnsafe adds a container to the map of containers being watched.
-func (engine *DockerStatsEngine) addContainerUnsafe(dockerID string) (*StatsContainer, error) {
+func (engine *DockerStatsEngine) addContainerUnsafe(dockerID string) (*StatsContainer, *StatsTask, error) {
 	// Make sure that this container belongs to a task and that the task
 	// is not terminal.
 	task, err := engine.resolver.ResolveTask(dockerID)
 	if err != nil {
-		return nil, errors.Wrapf(err, "could not map container to task, ignoring container: %s", dockerID)
+		return nil, nil, errors.Wrapf(err, "could not map container to task, ignoring container: %s", dockerID)
 	}
 
 	if len(task.Arn) == 0 || len(task.Family) == 0 {
-		return nil, errors.Errorf("stats add container: invalid task fields, arn: %s, familiy: %s", task.Arn, task.Family)
+		return nil, nil, errors.Errorf("stats add container: invalid task fields, arn: %s, familiy: %s", task.Arn, task.Family)
 	}
 
 	if task.GetKnownStatus().Terminal() {
-		return nil, errors.Errorf("stats add container: task is terminal, ignoring container: %s, task: %s", dockerID, task.Arn)
+		return nil, nil, errors.Errorf("stats add container: task is terminal, ignoring container: %s, task: %s", dockerID, task.Arn)
 	}
 
 	statsContainer, err := newStatsContainer(dockerID, engine.client, engine.resolver, engine.config)
 	if err != nil {
-		return nil, errors.Wrapf(err, "could not map docker container ID to container, ignoring container: %s", dockerID)
+		return nil, nil, errors.Wrapf(err, "could not map docker container ID to container, ignoring container: %s", dockerID)
 	}
 
 	seelog.Debugf("Adding container to stats watch list, id: %s, task: %s", dockerID, task.Arn)
 	engine.tasksToDefinitions[task.Arn] = &taskDefinition{family: task.Family, version: task.Version}
 
+	dockerContainer, errResolveContainer := engine.resolver.ResolveContainer(dockerID)
+	if errResolveContainer != nil {
+		seelog.Debugf("Could not map container ID to container, container: %s, err: %s", dockerID, err)
+	}
+
 	watchStatsContainer := false
+	var statsTaskContainer *StatsTask
 	if !engine.config.DisableMetrics {
 		// Adding container to the map for collecting stats
-		watchStatsContainer = engine.addToStatsContainerMapUnsafe(task.Arn, dockerID, statsContainer, engine.containerMetricsMapUnsafe)
+		watchStatsContainer = engine.addToStatsContainerMapUnsafe(task.Arn, dockerID, statsContainer,
+			engine.containerMetricsMapUnsafe)
+
+		if task.IsNetworkModeAWSVPC() {
```

Comment:
Can this be part of another method as it is becoming huge otherwise?

Reply:
Will make it a separate method.